p2p/module_test.go

```diff
@@ -0,0 +1,108 @@
+package p2p
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/defaults"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_configureBootstrapNodes(t *testing.T) {
+	defaultBootstrapNodes := strings.Split(defaults.DefaultP2PBootstrapNodesCsv, ",")
+	type args struct {
+		p2pCfg *configs.P2PConfig
+		m      *p2pModule
+	}
+	tests := []struct {
+		name               string
+		args               args
+		wantBootstrapNodes []string
+		wantErr            bool
+	}{
+		{
+			name: "unset boostrap nodes should yield no error and return DefaultP2PBootstrapNodes",
+			args: args{
+				p2pCfg: &configs.P2PConfig{},
+				m:      new(p2pModule),
+			},
+			wantErr:            false,
+			wantBootstrapNodes: defaultBootstrapNodes,
+		},
+		{
+			name: "empty string boostrap nodes should yield no error and return DefaultP2PBootstrapNodes",
+			args: args{
+				p2pCfg: &configs.P2PConfig{
+					BootstrapNodesCsv: "",
+				},
+				m: new(p2pModule),
+			},
+			wantErr:            false,
+			wantBootstrapNodes: defaultBootstrapNodes,
+		},
+		{
+			name: "untrimmed empty string boostrap nodes should yield no error and return DefaultP2PBootstrapNodes",
+			args: args{
+				p2pCfg: &configs.P2PConfig{
+					BootstrapNodesCsv: "     ",
+				},
+				m: new(p2pModule),
+			},
+			wantErr:            false,
+			wantBootstrapNodes: defaultBootstrapNodes,
+		},
+		{
+			name: "untrimmed string boostrap nodes should yield no error and return DefaultP2PBootstrapNodes",
+			args: args{
+				p2pCfg: &configs.P2PConfig{
+					BootstrapNodesCsv: "     http://somenode:50832  ",
+				},
+				m: new(p2pModule),
+			},
+			wantErr:            false,
+			wantBootstrapNodes: []string{"http://somenode:50832"},
+		},
+		{
+			name: "custom bootstrap nodes should yield no error and return the custom bootstrap node",
+			args: args{
+				p2pCfg: &configs.P2PConfig{
+					BootstrapNodesCsv: "http://somenode:50832,http://someothernode:50832",
+				},
+				m: new(p2pModule),
+			},
+			wantBootstrapNodes: []string{"http://somenode:50832", "http://someothernode:50832"},
+			wantErr:            false,
+		},
+		{
+			name: "malformed bootstrap nodes string should yield an error and return nil",
+			args: args{
+				p2pCfg: &configs.P2PConfig{
+					BootstrapNodesCsv: "\n\n",
+				},
+				m: &p2pModule{},
+			},
+			wantBootstrapNodes: []string(nil),
+			wantErr:            true,
+		},
+		{
+			name: "invalid hostname:port pattern for bootstrap nodes string should yield an error and return nil",
```

Comment:
Add details "port number too high yields an error and empty list of bootstrap nodes"

You can also easily add a couple tests for the wrong protocol and a negative port number (you have the code in place for it)